pinc/CharSuites.inc

```diff
@@ -178,3 +178,45 @@ foreach($charsuite_files as $charsuite_file)
 {
     include_once($charsuite_file);
 }
+
+function get_pickersets($charsuites)
+{
+    $pickersets = [];
+    foreach($charsuites as $charsuite)
+    {
+        $pickersets[] = $charsuite->pickerset;
+    }
+    return $pickersets;
+}
+
+function get_valid_codepoints($charsuites)
+{
+    # Codepoints applicable to all projects
+    $global_codepoints = [
+        'U+000A', # new lines
+        'U+000D', # carriage returns
+    ];
+
+    $codepoints = $global_codepoints;
+    foreach($charsuites as $charsuite)
+    {
+        $codepoints = array_merge($codepoints, $charsuite->codepoints);
+    }
+    return $codepoints;
+}
+
+function get_charsuites($identifier=NULL)
+{
+    try
+    {
+        $project = new Project($identifier);
+        return $project->get_charsuites();
+    }
+    catch (NonexistentProjectException $e)
+    {
```

Comment:
Since we're passing in "quiz" for the identifier (which I think is great) we should explicitly check against that:
```
    if($identifier == "quiz")
    {
        return [
            CharSuites::get('basic-latin'),
            CharSuites::get('extended-european-latin'),
        ];
    }
    else
    {
        $project = new Project($identifier);
        return $project->get_charsuites();
    }
```
Then we don't even need to add a comment of why we return something different if the project doesn't exist.

Reply:
Good idea. I've done this slightly differently.